### PR TITLE
Make registerShortcut from anki/shortcuts package override conflicting shortcuts

### PR DESCRIPTION
**Pull request by kleinerpirat**
Sparked by:
- https://github.com/kleinerpirat/anki-tooltips/issues/1

We discussed the Qt side in #2376 and came to the conclusion it's not worth the effort. But doing it on the JS side might be. This PR allows the following:

```ts
registerShortcut(
    (event: KeyboardEvent) => {
        // handle event
    },
    "Control+T",
    { event: "keydown", override: true},
);
```
As a result, all JS shortcuts (including emacs-style sequences) that conflict with the given keyCombination are disabled.

The example above would disable the LaTeX shortcuts:
 - Control+T, T
 - Control+T, E
 - Control+T, M

---
While it is (almost) reasonable to let add-on authors disable QShortcuts themselves, it is currently impossible to do the same in JS - how would they access the `remove` callback of native Svelte components that do not expose it?

### CODE REVIEW

**Comment by kleinerpirat** [comment]
`./ninja format` and `./ninja check` seem to have different opinions again.

**Comment by kleinerpirat** [comment]
An alternative to the `override` parameter would be to export a different function to the API than what our native components are using. Then add-on shortcuts would take precedence over native ones and existing add-ons would not have to be updated.

**Comment by hikaru-y** [comment]
At this time, there appear to be a few issues with this.

1. It is not possible to register multiple listeners to a single shortcut, which is currently possible.
    - An add-on may want to register multiple listeners with different event targets or different `AddEventListenerOptions` for the same shortcut.

1. If a `<Shortcut>` component is mounted after an add-on executes `registerShortcut()`, the shortcut key of that component cannot be overridden.
    1. Suppose an addon wants to override <kbd>Ctrl + Shift + C</kbd>.
    1. Open an editor with a non-Cloze note type, then change the note type to a Cloze one.
    1. Pressing that shortcut key will execute the original callback as well as the add-on's callback.

**Comment by kleinerpirat** [comment]
Thanks for testing this, Hikaru!

We could fix the first issue by simply not storing add-on shortcuts in the `registeredShortcuts` array. We should probably rename it to `nativeShortcuts` then.

The second issue is more involved.. But I feel like that could be solved on the add-on side. I'm at work the next 10 hours, so I won't be able to fix this today @dae.

**Comment by kleinerpirat** [comment]
@hikaru-y do you think it's reasonable to let add-on authors manually handle issue 2? I guess the other option is to store calls to `require("anki/shortcuts").registerShortcut` and reiterate the disabling process when the native `registerShortcut` is called later.

**Comment by dae** [comment]
(I'm tied up with AnkiWeb work at the moment, so I likely won't be able to get to the pending PRs until early next week I'm afraid)

**Comment by dae** [comment]
We've been focusing on *how can we*, but I'd like to take a step back and think about the *should we* first. From what I can tell, the example use case here is turning off the LaTeX shortcuts so that the add-on can use Ctrl+t instead. Do we really want to be encouraging add-ons to replace the default shortcuts with unrelated actions? This is potentially surprising/annoying for users, as it means they can no longer access built-in functionality easily when they install the add-on. Wouldn't it be better to pick an unused shortcut combination instead?

**Comment by dae** [comment]
(as an aside, I tried out your add-on for the first time today, and it's quite sleek)

**Comment by kleinerpirat** [comment]
If it were a fixed shortcut, I would simply fix the issue from the add-on side. But a recent update allows users to set their own shortcut for the tooltips.

So any time an add-on allows to set custom shortcuts, there will be trouble with conflicts. I wouldn't exactly call the PR "encouragement" to override native shortcuts, but I get why you're nervous about it.

Alternatively, we could provide a list of editor shortcuts for add-on authors to compare user input against and create an alert in case a user sets a conflicting one.

It's probably best to get some feedback on the issue from other devs.

**Comment by dae** [comment]
Ah, if the replacement is because the user deliberately chose a shortcut and not because it was the default, that makes more sense. One other option would be to tweak this code so that instead of disabling the existing shortcuts if found, it throws an error that the shortcut is already in use. The add-on could catch it, and report a message like "Ctrl+T is already used; please pick another shortcut". No strong feelings here. WDYT?

File: .eslintrc.js [hunk 1]
```diff
@@ -33,6 +33,8 @@ module.exports = {
             ],
             rules: {
                 "@typescript-eslint/no-non-null-assertion": "off",
+                "simple-import-sort/imports": "off",
+                "simple-import-sort/exports": "off",
```
Comment (dae) [inline .eslintrc.js:37]: This should not be necessary with the steps mentioned here: https://github.com/ankitects/anki/pull/2395#issuecomment-1445777657

Comment (kleinerpirat) [reply .eslintrc.js:37]: Thanks!

**Comment by kleinerpirat** [comment]
Throwing an error is a good idea! However, if a user doesn't care about a particular native shortcut, they may still want to override it. I'll revisit this later.

**Comment by dae** [comment]
Cleaning up old PRs - please open a new one if you'd like to continue this.

**Pull request closed without merging**